app/Services/ImportResults.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\ImportMeta;
+use App\Models\Mismatch;
+
+class ImportResults
+{
+    public function generateCSV(string $path, int $import_id): void
+    {
+        $mismatches = Mismatch::get();
+
+        // Creating the output stream
+        $handle = fopen($path, 'w');
+
+        $import = ImportMeta::find($import_id);
+        $mismatches = Mismatch::whereBelongsTo($import)->get();
+
+        $mismatchValues = array_map(function ($mismatch) {
+            return [
+                $mismatch['item_id'],
+                $mismatch['statement_guid'],
+                $mismatch['property_id'],
+                $mismatch['wikidata_value'],
+                $mismatch['meta_wikidata_value'],
+                $mismatch['external_value'],
+                $mismatch['external_url'],
+                $mismatch['review_status']
+            ];
+        }, $mismatches->toArray());
+
+        // Add column keys as CSV headers
+        $mismatchesWithKeys = array_merge([config('imports.results.column_keys')
```

Comment:
IMHO it would be nicer to emit the headers in a separate `fputcsv()` call (like in `ImportReport`), rather than merge two fairly different arrays together.